packages/bns/README.md

```diff
@@ -12,31 +12,278 @@ state without relying on any central points of control.
 ```
 npm install --save @stacks/bns
 ```
+## Example Usages
 
-## Example Usage
+### Check availability 
+Check if name can be registered
 
 ```typescript
 import { canRegisterName } from '@stacks/bns';
-import { StacksTestnet } from '@stacks/network';
-
-const network = new StacksMainnet();
+import { StacksTestnet, StacksMainnet } from '@stacks/network';
 
+// for mainnet, use `StacksMainnet()`
+const network = new StacksTestnet();
 const fullyQualifiedName = 'name.id';
+
 const result = await canRegisterName({ fullyQualifiedName, network });
+// true / false
 ```
 
+### Get name price 
+Get price of name registration in microstacks
+
 ```typescript
-import { buildRegisterNameTX } from '@stacks/bns';
-import { StacksTestnet } from '@stacks/network';
+import { getNamePrice } from '@stacks/bns';
+import { StacksTestnet, StacksMainnet } from '@stacks/network';
+
+// for mainnet, use `StacksMainnet()`
+const network = new StacksTestnet();
+const fullyQualifiedName = 'name.id';
+
+const price = await getNamePrice({ fullyQualifiedName, network });
+// price of name registration in microstacks
+```
 
-const network = new StacksMainnet();
+### Steps to register name
+Send two transaction to secure a name, preorder and then register.
 
-const name = 'name.id';
-const salt =  'example-salt'
-const zonefile =  'example-zonefile'
-const publicKey = 'SPF0324DSC4K505TP6A8C7GAK4R95E38TGNZP7RE'
+1. Preorder: Generates a name preorder transaction. First step in registering a name. This transaction does not reveal the name that is about to be registered. And it sets the amount of STX to be burned for the registration.
+2. Register: Generates a name transfer transaction. This changes the owner of the registered name.
+             
+### Preorder
+
+```typescript
+import { buildPreorderNameTx } from '@stacks/bns';
+import { StacksTestnet, StacksMainnet } from '@stacks/network';
+import {
+  TxBroadcastResult,
+  broadcastTransaction,
+  publicKeyToString,
+  TransactionSigner,
+  createStacksPrivateKey,
+  pubKeyfromPrivKey,
+} from '@stacks/transactions';
+
+// for mainnet, use `StacksMainnet()`
+const network = new StacksTestnet();
+const fullyQualifiedName = 'name.id';
+const salt = 'salt';
+const stxToBurn = 200n;
+const privateKey = 'cb3df38053d132895220b9ce471f6b676db5b9bf0b4adefb55f2118ece2478df01';
```

Comment:
I'm guessing this is fine for test code but we don't want to encourage people to hard-code in their private keys in production.

Is it worth putting in a caveat or providing some best-practices @reedrosenbluth ?

Reply:
yes, definitely!

Reply:
Added caveats for `privateKey `. Let me know if it's fine or need any further improvements.